### PR TITLE
Add Prioritized Approximation Loss feature

### PR DESCRIPTION
**Pull request by bilelsgh**
## Feature overview

Implementation of Prioritized Experience Replay (PER) with Prioritized Approximation Loss (PAL) (linked to https://github.com/DLR-RM/stable-baselines3/pull/1622).
A NeurIPS 2020 paper shows that using PER is equivalent to adapting the loss function while using uniform experience replay.

> The expected gradient of the loss function (1/τ) \* |δ(i)|^τ, where τ > 0, when used with PER, is equal to the expected gradient of the following loss under uniform sampling.
> [https://papers.neurips.cc/paper\_files/paper/2020/file/a3bf6e4db673b6449c2f7d13ee6ec9c0-Paper.pdf](https://papers.neurips.cc/paper_files/paper/2020/file/a3bf6e4db673b6449c2f7d13ee6ec9c0-Paper.pdf)

This means we can avoid managing a sorted buffer and the associated complexity, while still converging to the same gradient.

## Description

I've added a new loss function, which adapts the Huber Loss by incorporating priority as described in the referenced paper. The buffer itself performs uniform sampling (ReplayBuffer). Additionally, I implemented a PrioritizedReplayBuffer to initialize the parameters alpha and beta (following the PAL or PER papers) and to properly handle the case where the PAL Loss is applied within the training method.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes) 
In accordance with @AlexPasqua PR https://github.com/DLR-RM/stable-baselines3/pull/1622 (and the corresponding issue https://github.com/DLR-RM/stable-baselines3/issues/1242) (👋 @araffin )

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by nf2bi89** [comment]
Has anyone else been able to get this to work and provide a working example with the code they used? @bilelsgh does not appear to be active anymore. I've been getting a Runtimerror with element 0 does not require or have a grad_fn when using this implementation.

**Comment by bilelsgh** [comment]
> Has anyone else been able to get this to work and provide a working example with the code they used? @bilelsgh does not appear to be active anymore. I've been getting a Runtimerror with element 0 does not require or have a grad_fn when using this implementation.

Hi, I’m a bit busy these days, I didn’t have this error when i ran the code
I’ll try again next week and let u know if i have something similar

**Comment by nf2bi89** [comment]
@bilelsgh could you post the test code you used to run this implementation

**Comment by bilelsgh** [comment]
more complete and adapted PR here #2199
